rfd/0089-merge-webapps.md

```diff
@@ -0,0 +1,185 @@
+in order to be able to build the webapp assets and include into the final
+teleport binaries.
+
+## Process
+
+Below outlines the process for each repository in the order that the steps need
+to be taken. All repositories will be prepped for the final archival and merging
+but the final switch over steps will not be taken until v11 has been released.
+
+The git histories of each branch will be maintained while merging. [Merging repositories](https://stackoverflow.com/questions/13040958/merge-two-git-repositories-without-breaking-file-history)
+
+### Webapps repository
+
+[repository](https://github.com/gravitational/webapps)
+
+#### Actions
+
+- [ ] Triage issues.
+  - [ ] Close those that are no longer relevant.
+  - [ ] [Transfer remaining issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository) to the [Teleport repo](https://github.com/gravitational/teleport)
+  - [ ] Add the [ui label](https://github.com/gravitational/teleport/labels/ui) to
+        those transferred issues.
+- [ ] Triage PRs.
+  - [ ] Close those that are no longer relevant.
+  - [ ] Merge remaining PRs.
+  - [ ] Apply the necessary backports.
+
+### Webapps.e repository
+
+[repository](https://github.com/gravitational/webapps.e)
+
+#### Actions
+
+- [ ] Triage issues.
+  - [ ] Close those that are no longer relevant.
+  - [ ] [Transfer remaining issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository) to the [Teleport.e repo](https://github.com/gravitational/teleport.e)
+  - [ ] Add the [ui label](https://github.com/gravitational/teleport.e/labels/ui)
+- [ ] Triage PRs.
+  - [ ] Close those that are no longer relevant.
+  - [ ] Merge remaining PR's.
+  - [ ] Apply the necessary backports.
+
+### Teleport repository
+
+[repository](https://github.com/gravitational/teleport)
+
+#### Actions
+
+- [ ] Remove `/webassets` submodule
+  - This submodule is no longer required as the web UI will be built on demand.
+- [ ] Clone the [Webapps repository](https://github.com/gravitational/webapps) into
+      the Teleport root. [Maintaining their respective git histories](https://stackoverflow.com/questions/13040958/merge-two-git-repositories-without-breaking-file-history)
+  - [ ] This will need to be done for every respective version branch (v9, v10, v11)
+- [ ] Update targets that use the `packages/webapps.e` submodule to point points to
+      the correct version in the `teleport.e/webapps` folder.
+- [ ] Only require teleport build processes to run on teleport paths and the webapp
+      ones to run on the webapp paths
+-
+- Archive the [Webapps repository](https://github.com/gravitational/webapps).
+
+#### CI jobs
+
+- **CodeQL / Analyze (go) (pull_request)**
+- **Teleport-IntegrationTest (ci-account)**
+- **Teleport-UnitTest (ci-account)**
+- **Teleport-DocTest (ci-account)**
+- **Teleport-Lint (ci-account)**
```

Comment:
Any changes in these? Or are we keeping webapps build/lint checks contained in the webapps- jobs (which I would actually recommend)?

Reply:
No changes to these, the webapps jobs will be moved over and run separately. This will allow us to skip out of jobs that don't need to run.... No need to run the `Teleport-Lint` job for example when changes were only made to the UI.